Client/Frontend/Browser/TabManagerStore.swift

```diff
@@ -88,19 +89,28 @@ class TabManagerStore {
         assert(Thread.isMainThread)
         guard let savedTabs = prepareSavedTabs(fromTabs: tabs, selectedTab: selectedTab),
             let path = tabsStateArchivePath() else {
+                clearArchive()
                 return succeed()
         }
 
+        writeOperation.cancel()
+
         let tabStateData = NSMutableData()
         let archiver = NSKeyedArchiver(forWritingWith: tabStateData)
         archiver.encode(savedTabs, forKey: "tabs")
         archiver.finishEncoding()
+
         let result = Success()
-        DispatchQueue.global().async {
+        writeOperation = DispatchWorkItem {
             let written = tabStateData.write(toFile: path, atomically: true)
-            print("PreserveTabs: \(written)") // Ignore write failure (could be restoring).
+            log.debug("PreserveTabs write ok: \(written)") // Ignore write failure (could be restoring).
             result.fill(Maybe(success: ()))
         }
+
+        // Delay by 100ms to debounce repeated calls to preserveTabs in quick succession.
+        // Notice above that a repeated 'preserveTabs' call will 'cancel()' a pending write operation.
+        serialQueue.asyncAfter(deadline: .now() + 0.100, execute: writeOperation)
```

Comment:
Fix 2: put the work on a serial queue. The previous concurrent queue has no guarantee of order of work, two preserveTabs in quick succession could have resulted in the first call happening second.

Improvement: Guard the rate that preserveTabs can be called.